src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +451,338 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
                 iEndIndex = endIndex
             };
 
-            return expansion.InsertExpansion(textSpan, textSpan, this, LanguageServiceGuid, out ExpansionSession) == VSConstants.S_OK;
+            if (expansion.InsertExpansion(textSpan, textSpan, this, LanguageServiceGuid, out ExpansionSession) == VSConstants.S_OK)
+            {
+                // This expansion is not derived from a symbol
+                _symbols = default;
+                _symbol = null;
+                _arguments = _arguments.Clear();
+                return true;
+            }
+
+            if (!(SubjectBuffer.GetFeatureOnOffOption(CompletionOptions.EnableCallCompletionOnTabTab) ?? false))
+            {
+                // Full method call completion is not enabled
+                return false;
+            }
+
+            var symbols = ThreadingContext.JoinableTaskFactory.Run(() =>
+            {
+                var caretPosition = SubjectBuffer.CurrentSnapshot.GetPoint(endPositionInSubjectBuffer);
+                return GetSymbolsAsync(caretPosition, cancellationToken);
+            });
+
+            if (symbols.OfType<IMethodSymbol>().Any())
+            {
+                XNamespace snippetNamespace = "http://schemas.microsoft.com/VisualStudio/2005/CodeSnippet";
```

Comment:
nit: consider static field.

Reply:
➡️ Static field doesn't buy much here. Will extract both snippet creations to a helper method so it's at least in one place.